src/TwigComponent/doc/index.rst

```diff
@@ -951,6 +951,42 @@ When overriding the ``alert_message`` block, you have access to the ``message``
         {% endblock %}
     {% endcomponent %}
 
+Sometimes, you may want to override a block of a deeply nested component and you need access to
+some properties or functions from higher components. That can be done via the ``this.parent...`` hierarchy:
+
+.. code-block:: twig
+
+    {# templates/SuccessAlert.html.twig #}
+    {% component Alert with { type: 'success' } %}
+        {{ this.parent.someFunction }} {# this refers to SuccessAlert #}
+
+        {{ this.parent.someProp }} {# references a "someProp" prop from SuccessAlert #}
```

Comment:
Instead of `this.parent` (which requires us to use a wrapper `Hierarchy` object and magic methods), what about following our `outerBlocks` convention and adding an `outerContext` variable?

```
{{ outerContext.this.someProp }} 
```

Reply:
How would you refer this.parent.parent then?

Reply:
`outerContext.outerContext.this.someProp`? I think this would naturally occur if we started adding the `outerContext` variable, right?

Reply:
Changed and using `outerScope` variable instead of `outerContext` as discussed via Slack.